homeassistant/helpers/collection.py

```diff
@@ -327,13 +336,13 @@ def sync_entity_lifecycle(
     platform: str,
     entity_component: EntityComponent,
     collection: ObservableCollection,
-    create_entity: Callable[[dict], Entity],
+    create_entity: Callable[[dict], CollectionEntity],
```

Comment:
I think should be using ConfigType
```suggestion
    create_entity: Callable[[ConfigType], CollectionEntity],
```